### PR TITLE
feat: first draft to have basic skeleton for stress-example

### PR DESCRIPTION
**Pull request by theUnkownName**
- [x] Flow is similar to Pokedex. Encoder is MobileNet
- [x] At query time use Faiss with ref_indexer 
- [x] At query time use Annoy with ref_indexer 
- [x] At query time use Scann with ref_indexer 
- [x] Dummy random generator of image for input_fn.

### CODE REVIEW

File: stress-example/flows/index.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    uses: $ENCODER
+    parallel: $JINA_PARALLEL
+    timeout_ready: 600000
+    read_only: true
+  vec_idx:
+    uses: pods/vec.yml
+    shards: $JINA_SHARDS
+    separated_workspace: true
+    polling: all
+    uses_reducing: _merge_all
```
Comment (JoanFM) [inline stress-example/flows/index.yml:18]: just to keep things up to date, `_merge_all` to be changed to `_merge_matches`

File: stress-example/app.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+def config():
+    parallel = 1 if sys.argv[1] == 'index' else 1
+    shards = 1
+
+    os.environ['JINA_PARALLEL'] = str(parallel)
+    os.environ['JINA_SHARDS'] = str(shards)
+    os.environ['WORKDIR'] = './workspace'
+    os.makedirs(os.environ['WORKDIR'], exist_ok=True)
+    os.environ['JINA_PORT'] = os.environ.get('JINA_PORT', str(45678))
+    os.environ['ENCODER'] = os.environ.get('ENCODER', 'jinaai/hub.executors.encoders.image.torchvision-mobilenet_v2')
```
Comment (JoanFM) [inline stress-example/app.py:22]: this encoder does not exist anymore, I think using directly `ImageTorchEncoder` should be okey

File: stress-example/flows/index.yml (hunk 1 above)
Comment (theUnkownName) [reply stress-example/flows/index.yml:18]: Right, ill change it

File: stress-example/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# seems that BiT only works in 2.0.0 not 2.2.0, but hey it's google's problem!
+tensorflow_cpu==2.1.0
```
Comment (cristianmtr) [inline stress-example/requirements.txt:2]: So then should this be `2.0.0`?

Comment (JoanFM) [reply stress-example/requirements.txt:2]: we should use ImageTorchEncoder at the end

File: stress-example/app.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        exit(1)
+    if sys.argv[1] == 'index':
+        config()
+        workspace = os.environ['JINA_WORKSPACE']
+        if os.path.exists(workspace):
+            print(f'\n +---------------------------------------------------------------------------------+ \
+                    \n |                                   🤖🤖🤖                                        | \
+                    \n | The directory {workspace} already exists. Please remove it before indexing again. | \
+                    \n |                                   🤖🤖🤖                                        | \
+                    \n +---------------------------------------------------------------------------------+')
```
Comment (cristianmtr) [inline stress-example/app.py:64]: Make sure you do `sys.exit` here to force you to clean the workspace ;) To prevent the issue we just saw earlier

**Comment by JoanFM** [comment]
Hey @theUnkownName ,

remember to remove the workspace files and not push them.

Also, can you put this example in a subfolder (like image_search) so that we have a placeholder to later have the `nlp_search` example?

File: stress-example/image_search/app.py [hunk 5]
```diff
@@ -7,47 +7,44 @@
...
     import numpy as np
     return np.random.randint(0, 256, (img_height, img_width, 3))
 
+
 def random_docs():
     for idx in range(0, num_docs):
         with Document() as doc:
             doc.id = idx
-            doc.blob = create_random_img_array(3, 3)
+            doc.blob = create_random_img_array(25, 25)
```
Comment (JoanFM) [inline stress-example/image_search/app.py:22]: can u try with 224, 224 since it is more common?

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by theUnkownName** [comment]
Hey all, 
Here is the first version using:
- Annoy
- Shards: 1
- 
What you think? I had some problems with the separate workspace, but Joan pointed out that could be solved with the work that Florian was doing yesterday, so that could be the next step, to rebase that part and test with 2 shards?

File: stress-example/app.py (hunk 4 above)
Comment (cristianmtr) [reply stress-example/app.py]: What about this?

File: stress-example/image_search/app.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        exit(1)
+    if sys.argv[1] == 'index':
+        config()
+        workspace = os.environ['JINA_WORKSPACE']
+        if os.path.exists(workspace):
+            print(f'\n +---------------------------------------------------------------------------------+ \
+                    \n |                                   🤖🤖🤖                                        | \
+                    \n | The directory {workspace} already exists. Please remove it before indexing again. | \
+                    \n |                                   🤖🤖🤖                                        | \
+                    \n +---------------------------------------------------------------------------------+')
```
Comment (cristianmtr) [inline stress-example/image_search/app.py:71]: `sys.exit` will prevent the issue we had in the beginning

File: stress-example/image_search/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,3 @@
+torch
```
Comment (cristianmtr) [inline stress-example/image_search/requirements.txt:1]: specify versions

File: stress-example/image_search/app.py (hunk 6 above)
Comment (theUnkownName) [reply stress-example/image_search/app.py:71]: Got it

**Comment by theUnkownName** [comment]
It works now with Scann, Faiss and Annoy.

**Review by JoanFM — APPROVED** [review]

**Pull request merged**